library/alloc/src/rc.rs

```diff
@@ -403,7 +419,14 @@ impl<T> Rc<T> {
     /// impl Gadget {
     ///     /// Construct a reference counted Gadget.
     ///     fn new() -> Rc<Self> {
-    ///         Rc::new_cyclic(|me| Gadget { me: me.clone() })
+    ///         // `me` is an uninitialized `Weak` at the start of the closure.
```

Comment:
This makes it sound like the `Weak` itself is in an uninitialized state, which sounds more scary than necessary.

Maybe ``// `me` is a Weak<Gadget> pointing at the new allocation of the Rc we're constructing.``?